node/forest_libp2p/src/service.rs

```diff
@@ -63,57 +83,76 @@ impl Libp2pService {
             swarm.subscribe(topic);
         }
 
-        Libp2pService { swarm }
+        let (pubsub_sender_in, pubsub_receiver_in) = channel(20);
+        let (pubsub_sender_out, pubsub_receiver_out) = channel(20);
+        Libp2pService {
+            swarm,
+            pubsub_receiver_in,
+            pubsub_sender_in,
+            pubsub_receiver_out,
+            pubsub_sender_out,
+            log,
+        }
     }
-}
-
-impl Stream for Libp2pService {
-    type Item = NetworkEvent;
-    type Error = ();
 
-    /// Continuously polls the Libp2p swarm to get events
-    fn poll(&mut self) -> Result<Async<Option<Self::Item>>, Self::Error> {
+    /// Starts the `Libp2pService` networking stack. This Future resolves when shutdown occurs.
+    pub async fn run(self) -> Result<(), ()> {
```

Comment:
Is there a reason the return type is as such? I don't see the function returning anything?

Reply:
I had it in there before for some reason. Ill remove it